test/test_linalg.py

```diff
@@ -31,6 +32,125 @@ def fn(a, b):
         self.assertEqual(actual, original)
         self.assertEqual(actual, scripted)
 
+    def test_norm_dtype(self, device):
+        def run_test_case(input_size, ord, keepdim, from_dtype, to_dtype, compare_dtype):
+            msg = (
+                f'input_size={input_size}, ord={ord}, keepdim={keepdim}, '
+                f'from_dtype={from_dtype}, to_dtype={to_dtype}')
+            input = torch.randn(*input_size, dtype=from_dtype, device=device)
+            result = torch.linalg.norm(input, ord, keepdim=keepdim, dtype=from_dtype)
+            self.assertEqual(result.dtype, from_dtype, msg=msg)
+            result_converted = torch.linalg.norm(input, ord, keepdim=keepdim, dtype=to_dtype)
+            self.assertEqual(result_converted.dtype, to_dtype, msg=msg)
+            self.assertEqual(result.to(compare_dtype), result_converted.to(compare_dtype), msg=msg)
```

Comment:
Should this in general be true? If the dtype compares the computation type, then shouldn't computing in double sometimes be more precise than computing in float, for example?

Reply:
Right, double will be more precise. Because of that, if we instead promote the float Tensor up to double, the error bounds that `assertEqual` uses become slimmer (according to https://github.com/pytorch/pytorch/blob/master/torch/testing/_internal/common_utils.py#L952)--when I tried it this way, the assert would often fail. But if we demote the double to a float, like these tests currently do, I believe the error bounds should be wide enough to always pass the test. 

Although if there's a better way to do this test, I'm happy to change it.

Reply:
That makes sense.